lib/qesapdeployment.pm

```diff
@@ -163,6 +165,7 @@ sub qesap_pip_install {
     push(@log_files, $pip_install_log);
     record_info("QESAP repo", "Installing pip requirements");
     assert_script_run(join(' ', $pip_ints_cmd, '-r', $paths{deployment_dir} . '/requirements.txt | tee -a', $pip_install_log), 360);
+    script_run("deactivate");
```

Comment:
Is it preferred for a specific reason? I would use it, but in the relevant test functions in `t/09_qesapdeployment.t`, which has to `redefine` and mock functions like `script_run`, `script_run` is already redefined while `enter_cmd` is not. I used `script_run` to avoid adding more code on the tests. Should I use `enter_cmd` and add a redefinition for it in the tests?

(keep in mind that the tests only invoke the second deactivation, in `execute` - I kept the same function here simply for uniformity reasons. I could change this to `enter_cmd` and leave the execute line as `script_run` if we want).

Reply:
I'd say `enter_cmd` is a "lighter" function (`script_run()` does a lot more, which means there are more places for it to fail). I mainly use `script_run` when I want to do something with the retval, and `enter_cmd` if only sending a command to SUT without minding if it works or not. Main difference I can see is that `script_run` can kill your test if there are communication problems with the SUT, while `enter_cmd` will just type the command away and not fail even if the typed command does not reach the SUT.

What you say about `t/09_qesapdeployment.t` does make sense. So feel free to leave the call with `script_run` if you chose.